src/cluster_legacy.c

```diff
@@ -2738,7 +2745,7 @@ int clusterProcessPacket(clusterLink *link) {
 
         /* If there is extension data, which doesn't have a fixed length,
          * loop through them and validate the length of it now. */
-        if (hdr->mflags[0] & CLUSTERMSG_FLAG0_EXT_DATA) {
+        if (server.process_clustermsg_extensions && (hdr->mflags[0] & CLUSTERMSG_FLAG0_EXT_DATA)) {
```

Comment:
I don't think `server.process_clustermsg_extensions` is needed. The intent here is to emulate a non-extension capable **sender**. It is actually a good idea to harden the receiver logic so it could deal with such a sender. We should revert this line.